src/mlpack/core/math/columns_to_blocks.cpp

```diff
@@ -77,7 +77,7 @@ void ColumnsToBlocks::Transform(const arma::mat& maximalInputs,
       const size_t maxCol = j * colOffset + blockWidth;
 
       output.submat(minRow, minCol, maxRow, maxCol) =
-          arma::reshape(maximalInputs.col(k++), blockHeight, blockWidth);
+          arma::reshape(maximalInputs.col(++k), blockHeight, blockWidth);
```

Comment:
I think that this is the cause of https://dev.azure.com/mlpack/mlpack/_build/results?buildId=2671&view=logs&j=24d3abe3-ef0b-5deb-3aab-64d839de2c3c&t=b3d23cc5-b695-5043-0f03-2084bf2ff0b5&l=155

Reply:
Right, in this case, we need to use post-increment,